topic_resources/ada_cooks.md

```diff
@@ -1,71 +1,88 @@
+- a new rails 4.2.2 application.
+- rspec setup in document format (hint: use a .rspec config file)
+- code coverage reporting (hint: user simplecov)
 
+## User Stories
 
-## Extra Credit
-
-+ Add the ability to browse for Recipes by Ingredient or Gadget
-+ Code Metrics
-    + less than 20 issues reported by Rubocop
-    + no method with a Flog score > 30
-+ Add a feature that allows users to keep track of what Ingredients are in their physical kitchen. Add a SpiceRack, Pantry, Refrigerator, and Freezer model, and create a "has_many through" relationship. Ingredients should have a column that determines which location they belong in, and a quantity column should be on the "has_many through" relationship model.
-
-
-## Double Bonus Gold Star Territory
-+ Sign up for Code Climate (http://codeclimate.com) and add it to your project
-+ Add the CarrierWave gem (https://github.com/carrierwaveuploader/carrierwave) and allow the user to upload a picture for a Recipe or Ingredient
-+ Add the Devise gem (https://github.com/plataformatec/devise) to allow for multiple users, and for users to sign in or out
-+ Add an adminsection that allows Users who have the correct access permissions to edit the Cookbooks, Recipes, etc of all Users in the system
-
-
-## Evaluation
-
-You will be evaluated on:
-
-- Running `rspec` from Terminal with no failures
-- Code quality (is your code organized and logical)
-- Extra credit for additional methods to add neat functionality
-- Extra credit for writing additional specs
-- Extra credit for attractive or inventive user interface
-
+### As any User, I can...
+- search by the name of an Ingredient or Gadget and get related Recipes back.
+- __Recipes__
+  - view an alphabetical list of all Recipes
+  - view the details of a Recipe, inluding...
+    - a link to the User profile of the person who created the Recipe
+    - a link to the Gadget profile for any associated Gadgets
+    - links to the Ingridient profile for all associated Ingredients
+- __Ingredients__
+  - view an alphabetical list of all Ingedients
+  - view the details of an individual Ingredient, including...
+    - a list of links to Recipes associated with the Ingredient
+- __Gadgets__
+  - view an alphabetical list of all Gadgets
+  - view the details of one Gadget, including...
+    - a list of links to Recipes associated with the Gadget
+    
+### As an anonymous User I can...
+- register a new User account
+- Log In to an existing User account
+
+### As a Logged In User I can...
+- Log Out
+- __Cookbooks__
+  - See a list of Cookbooks I've already created, including...
+    - The number of Recipes associated with the Cookbook
+    - The number of unique Gadgets associated with the Cookbook
+    - The number of unique Recipes associated with the Cookbook
+  - See the details of a specific Cookbook, including...
+    - a list of links to the Recipes associated with the Cookbook
+    - a link to remove (unassociate) a Recipe from the Cookbook
```

Comment:
Add a note that the Recipe will still exist, it has just removed the association